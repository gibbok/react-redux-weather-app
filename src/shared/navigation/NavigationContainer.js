// @flow
import { connect } from 'react-redux'
import Navigations from './Navigations'
import { setActive } from './navigationActions'
import type { Dispatch } from 'redux' // eslint-disable-line no-unused-vars

const mapStateToProps = (state, ownProps) => {
  return {
  }
}

/* istanbul ignore next */
const mapDispatchToProps = (dispatch:Dispatch<*>, ownProps) => {
  return {
    onNavigationClick: (navigation) => {
      console.log('onNavigationClick')
      dispatch(setActive(navigation))
    }
  }
}

const NavigationContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Navigations)

export default NavigationContainer
