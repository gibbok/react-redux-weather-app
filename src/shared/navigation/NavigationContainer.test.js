import React from 'react'
import configureStore from 'redux-mock-store'
import NavigationContainer from './NavigationContainer'
import toJson from 'enzyme-to-json'
import {shallow} from 'enzyme'

describe('<NavigationContainer />', () => {
  it('should render', () => {
    const initialState = {
      navigationReducer: {
        navigation: {
          data: [
            {
              id: 'home',
              title: 'Weather',
              tooltip: 'Weather',
              isActive: true,
              url: '/'
            },
            {
              id: 'map',
              title: 'Map',
              tooltip: 'Weather map',
              isActive: false,
              url: '/map'
            }
          ],
          ui: {
            active: 'home'
          }
        }}
    }
    const mockStore = configureStore()
    const store = mockStore(initialState)
    const wrapper = shallow(<NavigationContainer store={store} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
