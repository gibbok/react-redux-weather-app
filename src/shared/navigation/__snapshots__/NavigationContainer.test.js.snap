// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationContainer /> should render 1`] = `
<withStyles(Navigations)
  navigations={
    Array [
      Object {
        "id": "home",
        "isActive": true,
        "title": "Weather",
        "tooltip": "Weather",
        "url": "/",
      },
      Object {
        "id": "map",
        "isActive": false,
        "title": "Map",
        "tooltip": "Weather map",
        "url": "/map",
      },
    ]
  }
  onNavigationClick={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
